legacy/core/src/main/java/com/fsck/k9/preferences/DefaultGeneralSettingsManager.kt

```diff
@@ -0,0 +1,151 @@
+@file:Suppress("DEPRECATION")
+
+package com.fsck.k9.preferences
+
+import com.fsck.k9.K9
+import com.fsck.k9.Preferences
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import net.thunderbird.core.preference.GeneralSettings
+import net.thunderbird.core.preference.GeneralSettingsManager
+import net.thunderbird.core.preference.PreferenceChangePublisher
+import net.thunderbird.core.preference.display.DisplaySettingsPreferenceManager
+import net.thunderbird.core.preference.network.NetworkSettingsPreferenceManager
+import net.thunderbird.core.preference.notification.NotificationPreferenceManager
+import net.thunderbird.core.preference.privacy.PrivacySettingsPreferenceManager
+import net.thunderbird.core.preference.storage.Storage
+
+/**
+ * Retrieve and modify general settings.
+ *
+ * Currently general settings are split between [K9] and [GeneralSettings]. The goal is to move everything over to
+ * [GeneralSettings] and get rid of [K9].
+ *
+ * The [GeneralSettings] instance managed by this class is updated with state from [K9] when [K9.saveSettingsAsync] is
+ * called.
+ */
+// TODO(#9432): Split GeneralSettings and GeneralSettingsManager in smaller classes/interfaces
+@Suppress("TooManyFunctions")
+internal class DefaultGeneralSettingsManager(
+    private val preferences: Preferences,
+    private val coroutineScope: CoroutineScope,
+    private val changePublisher: PreferenceChangePublisher,
+    private val privacySettingsPreferenceManager: PrivacySettingsPreferenceManager,
+    private val notificationPreferenceManager: NotificationPreferenceManager,
+    private val displaySettingsSettingsPreferenceManager: DisplaySettingsPreferenceManager,
+    private val networkSettingsPreferenceManager: NetworkSettingsPreferenceManager,
+    private val backgroundDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : GeneralSettingsManager {
+    val mutex = Mutex()
+
+    // TODO(#9232): Should be removed when K9 settings is completely migrated
+    // This fallback is required until we finalize the split of the GeneralSettings class and Manager.
+    // The GeneralSettings must be composed by other smaller Managers flows.
+    private val k9GeneralSettingsFallback = MutableStateFlow(value = loadGeneralSettings())
+
+    private val generalSettings = MutableStateFlow(value = loadGeneralSettings())
+        .combine(privacySettingsPreferenceManager.getConfigFlow()) { generalSettings, privacySettings ->
+            generalSettings.copy(
+                privacy = privacySettings,
+            )
+        }
```

Comment:
As there are no other dependencies to create the `GeneralSettings`, you don't need to create a `MutableStateFlow`. Instead, you could directly call the `PrivacySettingsPreferenceManager` and map its result to the `GeneralSettings` as shown below:

```suggestion
    private val generalSettings = privacySettingsPreferenceManager.getConfigFlow()
        .map { privacy ->
            GeneralSettings(privacy = privacy)
        }
```